Assets/Scripts/UI/Terminal/Pilots/PilotsUI.cs

```diff
@@ -1,177 +1,139 @@
-﻿using System.Collections;
-using System.Collections.Generic;
+﻿using System.Text;
 using UnityEngine;
 using UnityEngine.UI; 
 
 public class PilotsUI : MonoBehaviour
 {
 	public GameObject crewPanel;
-	public GameObject pilotButtonPrefab;
-	public PilotsContainer pilotsContainer;
+	public Transform scrollViewContent;
+	public GameObject crewItemPrefab;
+	public GameObject backButtonPrefab;
+
+	public GameObject pilotProfilePanel;
+	private Text pilotDetailsText; 
+	private Image pilotAvatar;
+	private Image shipAvatar; 
 
-	private GameObject pilotProfilePanel;
-	private Text pilotNameText;
-	private Text pilotDescriptionText; 
-	private Image pilotAvatar; 
+	public PilotsContainer pilotsContainer;
 
 	private void Awake()
 	{
-		GeneratePilotsUI(); 
+		GeneratePilotsUI();
+		TerminalUIManager.onTabButtonClicked += () => pilotProfilePanel.SetActive(false);
     }
 
 	private void GeneratePilotsUI()
     {
+		PopulateScrollView();
 		GeneratePilotProfilePanel();
-		GeneratePilotButtons();
-		GeneratePilotNameText();
-        GeneratePilotDescription();
-		GeneratePilotAvatar();
-		GenerateBackButton(); 
+        GenerateShipAvatar();
+        GeneratePilotAvatar();
+		GeneratePilotDetails();
+        GenerateBackButton();
+    }
+
+	private void PopulateScrollView()
+	{
+		foreach (Pilot pilot in pilotsContainer.pilots)
+		{
+            GameObject crewItem = Instantiate(crewItemPrefab, scrollViewContent);
+			crewItem.GetComponent<Button>().AddOnClick(() => OpenPilotProfilePanel(pilot));
+			crewItem.GetComponentInChildren<Text>().text = pilot.pilotName;
+		}
+	}
+
+	private void OpenPilotProfilePanel(Pilot pilot)
+	{
+		crewPanel.SetActive(false);
+		pilotProfilePanel.SetActive(true);
+		shipAvatar.sprite = pilot.ship.shipAvatar;
+		pilotAvatar.sprite = pilot.avatar;
+		pilotDetailsText.text = BuildDetailsString(pilot);
 	}
 
 	private void GeneratePilotProfilePanel()
 	{
 		pilotProfilePanel = new GameObject(PilotsConstants.profilePanelName);
 
-		// Sibling of the crew panel so that it can be active
-		// while the crew panel is inactive. 
+		// Sibling of the crew panel so that it can be active while the crew panel is inactive. 
 		pilotProfilePanel.transform.parent = crewPanel.transform.parent.transform;
 
 		RectTransform rectTransform = pilotProfilePanel.AddComponent<RectTransform>();
-		rectTransform.anchorMin = Vector2.zero;
```

Comment:
Revel in the amount of code that got deleted because of these extensions

Reply:
such a beaute